creator.go

```diff
@@ -0,0 +1,28 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package apmqueue
+
+import "context"
+
+// TopicCreator is an interface for creating topics.
+type TopicCreator interface {
+	// CreateTopics creates one or more topics.
+	//
+	// CreateTopics will leave existing topics untouched.
+	CreateTopics(ctx context.Context, topic ...Topic) error
```

Comment:
Should `DeleteTopics` also be defined in an interface?

Reply:
It's not needed at the moment. The code I intend to use it in will not be deleting topics, only creating.

I'll remove the `apmqueue.TopicCreator` interface for now, since it's not actually used anywhere in this PR.